### ISSUE TITLE
dartDefine is not working well with values containing white spaces

### ISSUE THREAD

**Issue by zacutti-basf** [hey24sheep/azure-flutter-tasks#89]
In the build task, when you pass an argument's value that contains white spaces, it get split incorrectly here:

![image](https://github.com/hey24sheep/azure-flutter-tasks/assets/131777248/4f9b1f31-1fdb-40cd-a744-4c872a99d8b9)

Probably you can get inspired with the Flutter's team solution to this issue, see here:
https://github.com/flutter/flutter-intellij/compare/master...devoncarew:flutter-intellij:fix_args_parsing
Basically they are using the ParametersListUtil.parse(..) method from the com.intellij.util.execution.ParametersListUtil package instead of dartDefineMulti.split(" ")

**Comment by hey24sheep** [comment]
Hi, thanks for the bug report.

That function is a closed source of IntelliJ but I think it could be bypassed with the `extraArgs=` by passing ` --dart-define-from-file`.

I am currently been very busy, but I will think of a solution to consider spaces between the values as well.

**Comment by zacutti-basf** [comment]
I'll use a workaround to avoid using spaces. Thank you!

**Comment by hey24sheep** [comment]
Test string "VALUE=foo bar character=Pinocchio id=12 name= hey 24 sheep role = sys    admin"

To fix this spacing issue, which could occur in `--extraArgs` and other flags as well. I think a `separatorArg` should be used with default value as "Space" (to support existing users, as it is currently the default value) if someone wants to use "space separated values" they should define the separator as well like ",".

**Comment by hey24sheep** [comment]
@zacutti-basf you can use the workaround. I will keep this issue open for next release. I am a little busy so can't promise the next release date. But, I will try to fix it whenever the next release comes.

**Comment by hey24sheep** [comment]
@zacutti-basf Hi, can you check if dart-define is working normally for you. It would really be helpful if you can validate and update me.

If this issue is fixed (so I can close this issue) on latest v0.3.13. Extension is already published on the marketplace.

**Comment by kightsonsanom** [comment]
@hey24sheep this is a breaking change which causes current workflows to fail. Could you please update README.md on how to use `dartDefineMulti` with latest changes?

**Comment by hey24sheep** [comment]
@kightsonsanom it shouldn't break, the working isn't changed but how the value is processed and passed down. It should work as normal. Your values can be like "var1=value var2=value 2"

If this isn't working please let me know.

Also if this isn't working that means Analayze task must be failing as well for past few years. Because this change is just duplicating what Analyze task is doing for past few years.

Is anyone else facing issues with latest update?

**Comment by kightsonsanom** [comment]
@hey24sheep yes, it stopped working for me, I had to rollback to 0.3.6 version. I am not using Analyze task so I can't compare this

**Comment by hey24sheep** [comment]
@kightsonsanom  thank you for logging the issue. Please give me a little time to fix it. I'll fix it at the earliest.

**Comment by zacutti-basf** [comment]
> @zacutti-basf Hi, can you check if dart-define is working normally for you. It would really be helpful if you can validate and update me.
> 
> If this issue is fixed (so I can close this issue) on latest v0.3.13. Extension is already published on the marketplace.

hi @hey24sheep, thank you a lot for your effort and very fast fix. Unfortunately in my organization I've to request the extension upgrade and could take a while to have it approved, and then in case we have an issue could take a while for the version rollback. I'll try to create a project and make some tests in a personal laptop to give you feedback asap.

**Comment by hey24sheep** [comment]
@zacutti-basf Hi, really appreciate that but latest update is failing (as updated by other users). I will make a fix within a day or two. So, you can test after that release for now I won't recommend upgrading to the latest version.

**Comment by ko16a46** [comment]
Hello, I also observed failing builds over the weekend, but I can confirm that the latest version 0.3.14 lets us build our app successfully with no further changes to our pipelines. Thanks @hey24sheep!

**Comment by hey24sheep** [comment]
@ko16a46 Thank you for confirming. Web builds were failing so I have updated the extension again to `0.3.15` now. It will also fix the `Test Task`.

@zacutti-basf You can use the new `argument separator` input param (check the readme). This way you can define your own input separator and provide values like "ARG1= hello world, arg2= something with spaces,".

You all can test and check your builds. If something is not working fine, comment here and I will try to fix that. Otherwise, I will close this issue within two days of no activity, thanks.

**Comment by hey24sheep** [comment]
Closing this issue as a new argument exists now `separator` which will solve white space issue. Use a different separator like `|` or `,`.

If this issue is still not solved, feel free to reopen this issue.
